llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9492,14 +9596,33 @@ void VPWidenMemoryInstructionRecipe::execute(VPTransformState &State) {
     }
   }
 
+  auto MaskValue = [&](unsigned Part) -> Value * {
```

Comment:
might be simpler to just move `        Value *MaskPart = isMaskRequired ? BlockInMaskParts[Part] : nullptr;` up outside the if?